server/src/main/java/org/apache/uniffle/server/DefaultFlushEventHandler.java

```diff
@@ -109,26 +168,33 @@ private void startEventProcessor() {
 
   protected void eventLoop() {
     while (!stopped && !Thread.currentThread().isInterrupted()) {
-      processNextEvent();
+      dispatchEvent();
     }
   }
 
-  protected void processNextEvent() {
+  protected void dispatchEvent() {
     try {
       ShuffleDataFlushEvent event = flushQueue.take();
       Storage storage = storageManager.selectStorage(event);
-      if (storage instanceof HadoopStorage) {
-        hadoopThreadPoolExecutor.execute(() -> handleEventAndUpdateMetrics(event, false));
-      } else if (storage instanceof LocalStorage) {
-        localFileThreadPoolExecutor.execute(() -> handleEventAndUpdateMetrics(event, true));
+
+      Executor dedicatedExecutor = fallbackThreadPoolExecutor;
+      // pending event will be delegated to fallback threadPool
+      if (!event.isPended()) {
+        if (storage instanceof HadoopStorage) {
+          dedicatedExecutor = hadoopThreadPoolExecutor;
+          ShuffleServerMetrics.gaugeHadoopFlushThreadPoolQueueSize.inc();
+        } else if (storage instanceof LocalStorage) {
+          dedicatedExecutor = localFileThreadPoolExecutor;
+          ShuffleServerMetrics.gaugeLocalfileFlushThreadPoolQueueSize.inc();
+        }
       } else {
-        // When we did not select storage for this event, we will ignore this event.
-        // Then we must doCleanup, or will result to resource leak.
-        fallbackThreadPoolExecutor.execute(() -> event.doCleanup());
```

Comment:
The logic of this line is missed?

Reply:
Needn't. It will also dispatch event into consumer that will unify all logic